src/types.ts

```diff
@@ -0,0 +1,352 @@
+export interface User {
```

Comment:
the split was kinda dependent on if they were shared between files. `Repository`, `Sender`, etc are used all over, but Repo is only returned from one func so I just put it near it's function. I'm not sure what a better rule of thumb is.